src/borg/testsuite/helpers.py

```diff
@@ -104,6 +104,21 @@ def test_no_slashes(self, monkeypatch):
         with pytest.raises(ValueError):
             Location('/some/path/to/repo::archive_name_with/slashes/is_invalid')
 
+    def test_only_printable_characters(self, monkeypatch):
```

Comment:
the utility function then could be tested on its own, disconnected from the archivename context.